sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -462,8 +462,10 @@ case class DataSource(
     val allPaths = paths ++ caseInsensitiveOptions.get("path")
     val outputPath = if (allPaths.length == 1) {
       makeQualified(new Path(allPaths.head))
-    } else {
+    } else if (allPaths.length > 1) {
       throw QueryExecutionErrors.multiplePathsSpecifiedError(allPaths)
+    } else {
+      throw QueryExecutionErrors.dataPathNotSpecifiedError()
```

Comment:
Do you know if there's any test case that can cover the new branch? If not, could you add one in this pr? @vrozov